cmd-ls.c

```diff
@@ -373,11 +373,14 @@ int cmd_ls(int argc, char **argv)
 	session_free(session);
 	blob_free(blob);
 
+  /* NB: valgrind identified something related to account_array as a leak, though
+   * freeing the entries crashes the cli.
```

Comment:
The reason this crashes is that most entries in the account_array[i] are just a copy of the account already stored in blob's account list; so the account freeing that happens in blob_free() ends up as a double free.

The leak is that _some_ of the accounts in account_array (the shares) are bogus creations just so qsort() will do something sensible.  So, we need to identify those and free only those.